addons/web/static/src/views/form/form_error_dialog/form_error_dialog.js

```diff
@@ -6,18 +6,29 @@ import { useService } from "@web/core/utils/hooks";
 import { Component } from "@odoo/owl";
 
 export class FormErrorDialog extends Component {
+    static props = {
+        message: { type: String, optional: true },
+        data: { type: Object },
+        onDiscard: Function,
+        onStayHere: Function,
+        close: Function,
+    };
     setup() {
         this.action = useService("action");
         this.message = this.props.message;
         if (this.props.data.name === "odoo.exceptions.RedirectWarning") {
             this.message = this.props.data.arguments[0];
             this.redirectAction = this.props.data.arguments[1];
             this.redirectBtnLabel = this.props.data.arguments[2];
+            this.additionalContext = this.props.data.arguments[3];
         }
     }
 
-    onRedirectBtnClicked() {
-        this.action.doAction(this.redirectAction);
+    async onRedirectBtnClicked() {
+        await this.action.doAction(this.redirectAction, {
+            additionalContext: this.additionalContext,
+        });
+        this.stay();
```

Comment:
isn't it rather discard? We're leaving so...

Reply:
Actually we don't know, the action could be target new

Reply:
> Actually we don't know, the action could be target new

exactly